foundation/foundation-mda/src/main/resources/templates/data-delivery-pyspark/persist.py.vm

```diff
@@ -19,20 +19,33 @@
           SQLite: 'sqlite:///db.sqlite'
       :return: A connection object from the RDBMS connection pool.
       """
-      connection = self.rdbms_connection_pool.get_connection(db_url)
-      return connection
-    #end
+      return self._get_spark_connection(db_url)
+
+    def _get_spark_connection(self, name: str):
+        if self.session is not None:
+            return self.session
+        else:
+            self.session = self.create_spark_session(name)
+            return self.session
+
+   #end
 
    #if ($step.persist.shortCollectionType)
     def save_dataset(self, dataset: ${step.persist.shortCollectionType}[${step.persist.shortRecordType}], ${saveParam}) -> None:
    #elseif ($step.isPersistTypeRdbms())
     #set ($db_url = "db_url: str")
     def save_dataset(self, ${db_url}, dataset: ${step.persist.shortRecordType}, ${saveParam}) -> None:
       ${step.capitalizedName}Base.logger.info('Saving %s To RDBMS...' % self.descriptive_label)
-      rdbms_session = self.rdbms_connection_pool.get_connection(db_url)
-      engine = rdbms_session.get_bind()
-      dataset.createOrReplaceTempView(table_name)
-      rdbms_session.close()
+      rdbms_session = self.get_connection(db_url)
+      jdbc_properties = {
```

Comment:
S: It would be ideal if JDBC properties and other [JDBC data source options](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html#data-source-option) were exposed in such a way that they could be easily extended and/or augmented within pipeline implementations.  Even just breaking out connection option creation into a separate method would work:

```
def _get_jdbc_data_source_connection_options(self) -> dict:
  return {
            "url": ...,
            ...
  }
```

Reply:
After pruning the use of SQLAlchemy connection pooling, the jbasic dbc connection properties can now just be directly configured through krausening properties (spark-rdbms.properties) for any relational database. Additionally, the class `get_data_source_connection` can be overwritten in the impl to provide additional configs unique to specific kinds of databases.